Add freeipa (not containerised) (#241)
stackhpc/ansible-slurm-appliance: ansible/roles/freeipa/README.md

# freeipa

Support FreeIPA in the appliance. In production use it is expected the FreeIPA server(s) will be external to the cluster, implying that hosts and users are managed outside the appliance. However for testing and development the role can also deploy an "in-appliance" FreeIPA server, add hosts to it and manage users in FreeIPA.

# FreeIPA Client

## Usage
- Add hosts to the `freeipa_client` group and run (at a minimum) the `ansible/iam.yml` playbook.
- Host names must match the domain name. By default (using the skeleton Terraform) hostnames are of the form `nodename.cluster_name.cluster_domain_suffix` where `cluster_name` and `cluster_domain_suffix` are Terraform variables.

> - Hosts discover the FreeIPA server FQDN (and their own domain) from DNS records. If this is not set from DHCP, then use the `resolv_conf` role to configure this. For example when using the in-appliance FreeIPA development server,:

Reviewer: ```suggestion
- Hosts discover the FreeIPA server FQDN (and their own domain) from DNS records. If DNS servers are not set from correctly by DHCP, use the `resolv_conf` role to configure this. For example when using the in-appliance FreeIPA development server,:
```

  
  ```ini
  # environments/<env>/groups
  ...
  [resolv_conf:children]
  freeipa_client
  ...
  ```

  ```yaml
  # environments/<env>/inventory/group_vars/all/resolv_conf.yml
  resolv_conf_nameservers:
  - "{{ hostvars[groups['freeipa_server'] | first].ansible_host }}"
  ```

> - For production use with an external FreeIPA server, a random one-time password (OTP) must be generated when adding hosts to FreeIPA (e.g. using `ipa host-add --random ...`). This password should be set as a hostvar `freeipa_host_password`. Initial host enrolment will use this OTP to enrole the host. After this it becomes irrelevant so it does not need to be committed to git. This approach means the appliance does not require the FreeIPA administrator password.

Reviewer: ```suggestion
- For production use with an external FreeIPA server, a random one-time password (OTP) must be generated when adding hosts to FreeIPA (e.g. using `ipa host-add --random ...`). This password should be set as a hostvar `freeipa_host_password`. Initial host enrolment will use this OTP to enrol the host. After this, it becomes irrelevant so it does not need to be committed to git. This approach means the appliance does not require the FreeIPA administrator password.
```
